File: java/src/main/java/ai/buble/sdk/chat/ChatService.java
package ai.buble.sdk.chat;

import ai.buble.sdk.http.BubleHttpClient;

/**
 * Chat model discovery and protocol-compatible chat methods.
 */
public final class ChatService {
    private final ChatModelsService models;
    private final ChatCompletionsService completions;
    private final MessagesService messages;
    private final GeminiService gemini;

    public ChatService(BubleHttpClient http) {
        this.models = new ChatModelsService(http);
        this.completions = new ChatCompletionsService(http);
        this.messages = new MessagesService(http);
        this.gemini = new GeminiService(http);
    }

    public ChatModelsService models() {
        return models;
    }

    public ChatCompletionsService completions() {
        return completions;
    }

    public MessagesService messages() {
        return messages;
    }

    public GeminiService gemini() {
        return gemini;
    }
}

File: dotnet/src/Buble.Sdk/Chat/ChatService.cs
using Buble.Sdk.Http;

namespace Buble.Sdk.Chat;

public sealed class ChatService
{
    internal ChatService(BubleHttpClient http)
    {
        Models = new ChatModelsService(http);
        Completions = new ChatCompletionsService(http);
        Messages = new MessagesService(http);
        Gemini = new GeminiService(http);
    }

    public ChatModelsService Models { get; }

    public ChatCompletionsService Completions { get; }

    public MessagesService Messages { get; }

    public GeminiService Gemini { get; }
}
